### PR TITLE
serde in no_std

### PR DESCRIPTION
**Pull request by kvinwang**
This PR adds the basic serde support in no_std.

### CODE REVIEW

File: ethabi/Cargo.toml [hunk 1]
```diff
@@ -38,16 +39,20 @@ std = [
 	"sha3/std",
 	"ethereum-types/std",
 	"thiserror",
-	"uint/std",
+	"uint?/std",
```
Comment (nlordell) [inline ethabi/Cargo.toml:42]: Should we also add:
```suggestion
	"uint?/std",
	"serde?/std",
```

File: ethabi/src/filter.rs [hunk 2]
```diff
@@ -121,21 +119,20 @@ impl<T> From<Topic<T>> for Vec<T> {
...
 	{
-		let value = match *self {
-			Topic::Any => Value::Null,
+		match *self {
+			Topic::Any => Option::<()>::None.serialize(serializer),
 			Topic::OneOf(ref vec) => {
-				let v = vec.iter().map(|h| format!("0x{:x}", h)).map(Value::String).collect();
-				Value::Array(v)
+				let v: Vec<_> = vec.iter().map(|h| format!("0x{:x}", h)).collect();
+				v.serialize(serializer)
```
Comment (nlordell) [inline ethabi/src/filter.rs:132]: AFAIK `Hash: Serialize` - so you should be able to:
```suggestion
				vec.serialize(serializer)
```

File: ethabi/src/filter.rs [hunk 3]
```diff
@@ -121,21 +119,20 @@ impl<T> From<Topic<T>> for Vec<T> {
...
 			}
-			Topic::This(ref hash) => Value::String(format!("0x{:x}", hash)),
-		};
-		value.serialize(serializer)
+			Topic::This(ref hash) => format!("0x{:x}", hash).serialize(serializer),
```
Comment (nlordell) [inline ethabi/src/filter.rs:134]: Same comment here:
```suggestion
			Topic::This(ref hash) => hash.serialize(serializer),
```

**Review by nlordell — APPROVED** [review]
Changes look good to me.

File: ethabi/src/filter.rs (hunk 3 above)
Comment (kvinwang) [reply ethabi/src/filter.rs]: Done

**Comment by kvinwang** [comment]
@nlordell Your suggestions applied. Thanks.

**Comment by kvinwang** [comment]
Rebased onto master

**Review by vkgnosis — APPROVED** [review]

**Comment by vkgnosis** [comment]
Looks good to me. Will let it sit for one more day and merge if no other comments.

**Pull request merged**